### DISCUSSION TITLE
Transparent dolphin view.

### DISCUSSION THREAD

**Discussion by edpil02** [paulmcauley/klassy#15] [Ideas]
Is it possible to add a tweak like Lightly or kvantum for  having transparency and blur  in dolphin view
Thanks.

**Comment by paulmcauley** [comment]
Hi, the latest code on master (due to be released in v4.0) now supports transparency and blur in the entire Header area. For example:
![translucent_headers_with_blur](https://user-images.githubusercontent.com/6924479/173200596-3bfbeca2-d2d7-4ce8-a0e9-b46128928953.png)

I'm not sure though I want to start adding hacks for individual applications.

**Comment by edpil02** [comment]
Thanks for your job.
I agree that hacking individual applications is not a solution but it seems that dolphin transparency is specific because the applications
style i used before Classik have a option for dolphin transparent view ( columns and background).  
![dolphinView](https://user-images.githubusercontent.com/43163048/173310257-b83b4683-09a2-4607-b89e-31102713ce83.png)

**Comment by edpil02** [comment]
Hi i just sent a little paypal to tanks you.

**Reply by paulmcauley** [reply]
Merci beaucoup! Much appreciated

**Comment by paulmcauley** [comment]
Do you know if the Lightly transparent header feature is still working, or how to make it work? It doesn't work for me. I wanted to compare code with Lightly to see if I can iron-out a few glitches with the transparent headers.

Bug testing on the current master is much welcomed.

**Comment by edpil02** [comment]
Although I'm not a dev  nor a programmer a had to touch  (with difficulty )  Lightly code  to make it work properly (with dolphin total transparency).  
https://github.com/Luwx/Lightly/issues/98

I  dont use Lightly anymore , i use classik  but it seems that  Lighty code is  broken  with plasma 5.25  (issues 172 -171).
 If I understood correctly , dolphin hack is in     /kstyle/lightlystyle.cpp
Sorry for my english.

**Reply by adam-adrian** [reply]
I think the title bar/window decoration in lightly is not blurred because they not implement the new setBlurRegion API yet. And there is already a pull request that fix the window decoration setting issue (PR 177).

**Comment by edpil02** [comment]
Dolphin transparent  view  piece of code from  Kvantum.

  if (isDolphin_
      && qobject_cast<QAbstractScrollArea*>(getParent(widget,2))
      && !qobject_cast<QAbstractScrollArea*>(getParent(widget,3)))
  {
    /* Dolphin sets the background of its KItemListContainer's viewport
       to KColorScheme::View (-> kde-baseapps -> dolphinview.cpp).
       We force our base color here. */
    QPalette palette = widget->palette();
    palette.setColor(QPalette::Active, widget->backgroundRole(),
                     standardPalette().color(QPalette::Active,QPalette::Base));
    palette.setColor(QPalette::Inactive, widget->backgroundRole(),
                     standardPalette().color(QPalette::Inactive,QPalette::Base));
    forcePalette(widget, palette);

    /* hack Dolphin's view */
    if (hspec_.transparent_dolphin_view && widget->autoFillBackground())
      widget->setAutoFillBackground(false);

**Comment by edpil02** [comment]
I made some progress with  klassy and  dolphin hack view . 
![Screenshot_20230530_143714](https://github.com/paulmcauley/klassy/assets/43163048/b0149b08-0e7b-4ad8-86d2-464363712730)

**Reply by paulmcauley** [reply]
Did you get it to work? Do you have your source-code shared somewhere online?

I want to add App-Specific Overrides (exceptions) and Presets features to the Application Style first (which will be a lot of work), then will maybe look at adding things like this.

**Comment by edpil02** [comment]
Hi.
I made a port of klassy in Qt6 and imported all the necessary code from Lightly in a patch.
When transparent dolphin view is checked  in the  UI  , the background is changed but transparency is missing.
  
Seems that  Lightly and Klassy do not handle translucency in the same way.
I asked you for a little help in a bug report but you closed it, telling me that you would see it later.
https://github.com/paulmcauley/klassy/issues/91
So I gave up.
I'll see if I still have my patches.
Thanks

**Reply by paulmcauley** [reply]
#91 was closed because "thin line under toolbar" was a duplicate of another issue, not because of this translucency issue.

I haven't looked at the Lightly code yet, but a difference between Lightly and Klassy translucency I can think of is that Lightly reads the translucency from the colour-scheme only, whereas Klassy can also add it in. For translucency to work you also have to set the "Qt::WA_TranslucentBackground" attribute on the main window (https://github.com/paulmcauley/klassy/blob/c42d0183eb67453c31dcaafbd9fd629738ca71f2/kstyle/breezetoolsareamanager.cpp#L59C1-L59C63), and make sure there is nothing else painted behind what you are trying to make transparent.

**Comment by edpil02** [comment]
Here are the first patches I made for"dolphin view"  for klassy.
I no longer have the recent ones because as I am on plasma 6, I use lightly and klassy that i've  ported to  KF6.
But for klassy I 've   only   ported   and use the decorations.
I will look to see if there is any missing code left, but for your information in the lightly code, it is specified that this code was copied from kvantum which has this option.
I think the kvantum developer could be more useful to you than me.
[patchs.zip](https://github.com/paulmcauley/klassy/files/13980329/patchs.zip)

**Comment by edpil02** [comment]
Hi.
Are  you planning on making a klassy port  to qt6 as I'm going to try to get back into it.
Thanks.

**Reply by paulmcauley** [reply]
Yes, I plan a Qt6 release to coincide with the Plasma 6 release. I'm adding a lot of features first to the Plasma 5 release so that the features are also available on Plasma 5.27 LTS.
